### PR TITLE
ENH - new download handling for observation importer.

### PR DESCRIPTION
**Pull request by rksh**
streaming JSON, some cleanups with auth. need to make sure that these are all kosher.

### CODE REVIEW

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/DataFragment.java [hunk 1]
```diff
@@ -239,18 +239,23 @@ public void onClick( final View buttonView ) {
...
-        if (MainActivity.getMainActivity().isTransferring()) {
+        final SharedPreferences prefs = getActivity().getSharedPreferences(ListFragment.SHARED_PREFS, 0);
+        final String authname = prefs.getString(ListFragment.PREF_AUTHNAME, null);
+
+        if (null == authname) {
             importObservedButton.setEnabled(false);
+        } else if (MainActivity.getMainActivity().isTransferring()) {
+                importObservedButton.setEnabled(false);
+        } else {
+            importObservedButton.setOnClickListener(new OnClickListener() {
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/DataFragment.java]: Pry would leave the listener outside the else, so it's always set, so if the button is enabled things will work.

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/ApiDownloader.java [hunk 2]
```diff
@@ -33,11 +34,10 @@ protected void subRun() throws IOException, InterruptedException, WiGLEAuthExcep
             }
             final JSONObject json = new JSONObject(result);
             listener.requestComplete(json, false);
-        }
-        catch (final WiGLEAuthException waex) {
+        } catch (final WiGLEAuthException waex) {
             // ALIBI: allow auth exception through
             throw waex;
-        } catch (final Exception ex) {
+        } catch (final JSONException ex) {
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/ApiDownloader.java]: Is there a reason to reduce scope of exception? hopefully handled higher up?

**Comment by bobzilladev** [comment]
yay streaming! lgtm, only minor comments.

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/DataFragment.java (hunk 1 above)
Comment (rksh) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/DataFragment.java]: roger that.

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/ApiDownloader.java (hunk 2 above)
Comment (rksh) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/ApiDownloader.java]: general purpose exception handler was preventing error bubbling up - really just needed JSONException handling at this level

**Comment by bobzilladev** [comment]
+1

**Pull request merged**